client/verta/verta/tracking/entities/_deployable_entity.py

```diff
@@ -146,6 +146,100 @@ def download_model(self, download_to_path):
         """
         raise NotImplementedError
 
+    @staticmethod
+    def _build_artifact_store_path(artifact_stream, key, ext=None):
+        """Build a path for Verta's backend artifact store to store `artifact_stream`.
+
+        Parameters
+        ----------
+        artifact_stream : file-like
+            Artifact bytes.
+        key : str
+            Artifact key.
+        ext : str, optional
+            File extension associated with `artifact_stream`.
+
+        Returns
+        -------
+        str
+            Artifact store path for `artifact_stream`
+
+        """
+        # calculate checksum
+        checksum = _artifact_utils.calc_sha256(artifact_stream)
+        artifact_stream.seek(0)
+
+        # determine "file"name
+        #     The key might already contain the file extension, thanks to our hard-coded deployment
+        #     keys e.g. "model.pkl" and "model_api.json".
+        if ext is None or key.endswith("." + ext):
+            filename = key
+        else:
+            filename = key + "." + ext
+
+        return checksum + "/" + filename
+
+    @classmethod
+    def _create_artifact_msg(
+        cls,
+        key,
+        artifact_stream,
+        artifact_type,
+        method,
+        framework=None,
+        extension=None,
```

Comment:
This is more parameters than I'd like in a method, and I'm hoping to address it in the near-future with https://github.com/VertaAI/docs/pull/68